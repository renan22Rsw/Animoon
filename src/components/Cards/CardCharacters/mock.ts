interface mockTypes {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

export const mock: mockTypes = {
  id: 1,
  name: "Naruto",
  image: "https://example.com/character-image.jpg",
  role: "Main",
  voices: [
    {
      id: 1,
      name: "Junko Takeuch",
      image: "https://example.com/voices-image.jpg",
      language: "Japanease",
    },
  ],
};
