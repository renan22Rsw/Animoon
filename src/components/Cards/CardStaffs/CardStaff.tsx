import Image from "next/image";
import Link from "next/link";
import React from "react";

interface StaffProps {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

const CardStaff = ({ id, name, image, occupation }: StaffProps) => {
  return (
    <div className=" bg-[#1C1C1C] h-[80px]  w-[350px] m-4 rounded-md lg:w-[381px] 2xl:w-[380px]  2xl:h-[70px]">
      <Link href={`/staffs/${id}`} className="flex items-center text-xs">
        <Image
          className="h-20 w-[60px] rounded-sm 2xl:h-[70px] 2xl:w-[55px] "
          src={image}
          alt="Card-image-staffs"
          width={50}
          height={60}
          priority={true}
        />
        <span className="grid p-2 gap-y-6  grid-rows-2 items-center">
          <h4>{name}</h4>
          <p>{occupation + ""}</p>
        </span>
      </Link>
    </div>
  );
};

export default CardStaff;
