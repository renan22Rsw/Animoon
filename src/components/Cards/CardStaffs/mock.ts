interface mockTypes {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

export const mock: mockTypes = {
  id: 1,
  name: "Masashi Kishimoto",
  image: "https://example.com/staff-image.jpg",
  occupation: "Original Creator",
};
