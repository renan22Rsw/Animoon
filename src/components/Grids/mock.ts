interface mockAnimesAndMangas {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface mockCharacter {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

export const mockAnimesAndMangasMainPage: mockAnimesAndMangas = {
  id: 1,
  title: { romaji: "Akame ga kill" },
  coverImage: { large: "https://example.com/anime-image.jpg" },
};

export const mockCharacterMainPage: mockCharacter = {
  id: 2,
  name: { full: "Akame" },
  image: { large: "https://example.com/character-image.jpg" },
};
