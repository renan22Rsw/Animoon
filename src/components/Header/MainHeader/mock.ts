interface mockTypes {
  title: string;
  description: string;
  coverImage: string;
}

export const mock: mockTypes = {
  title: "Naruto",
  description:
    "Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.",
  coverImage: "https://example.com/image.jpg",
};
