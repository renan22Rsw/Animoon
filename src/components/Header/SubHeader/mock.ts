interface mockTypes {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown: string;
  yearsActive?: number[];
  description: string;
}

export const mock: mockTypes = {
  name: "Naruto",
  nativeName: "Naruto Uzumaki",
  alternativeName: ["Nine-Tails Jinchuuriki", " Number One Unpredictable"],
  age: "17",
  image: "http://example.com/image.jpg",
  gender: "Male",
  month: 10,
  day: 10,
  bloodType: "B",
  hometown: "Saitama Prefecture, Japan",
  yearsActive: [1996],
  description: "Born in Konohagakure, a ninja village hidden in the leaves",
};
