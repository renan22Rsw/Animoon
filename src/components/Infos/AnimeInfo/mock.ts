interface mockTypes {
  format: string;
  duration: number;
  status: string;
  seasonYear: number;
  season: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  duration: 24,
  averageScore: 85,
  meanScore: 88,
  favourites: 500,
  format: "TV",
  genres: ["comedy", "horror"],
  popularity: 1000,
  season: "winter",
  seasonYear: 2024,
  source: "Light Novel",
  status: "Airing",
};
