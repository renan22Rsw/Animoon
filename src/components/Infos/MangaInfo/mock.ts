interface mockTypes {
  format: string;
  status: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  averageScore: 80,
  meanScore: 85,
  favourites: 1200,
  format: "Manga",
  genres: ["action", "adventure"],
  popularity: 1000,
  source: "Original",
  status: "Releasing",
};
