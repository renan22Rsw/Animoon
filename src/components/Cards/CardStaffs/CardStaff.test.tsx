import { render, screen } from "@testing-library/react";
import CardStaff from "./CardStaff";
import { mock } from "./mock";

describe("CardStaff component", () => {
  it("should render CardStaff images", () => {
    render(<CardStaff {...mock} />);
    const staffImage = screen.getByRole("img");
    expect(staffImage).toBeInTheDocument();
    expect(staffImage).toHaveAttribute("width", "50");
    expect(staffImage).toHaveAttribute("height", "60");
    expect(staffImage).toHaveAttribute("fetchpriority", "high");
  });

  it("should render CardStaff links", () => {
    render(<CardStaff {...mock} />);
    const staffLink = screen.getByRole("link");
    expect(staffLink).toBeInTheDocument();
    expect(staffLink).toHaveAttribute("href", `/staffs/${mock.id}`);
  });

  it("should render titles of CardStaff", () => {
    render(<CardStaff {...mock} />);
    expect(screen.getByText(mock.name)).toBeInTheDocument();
    expect(screen.getByText(mock.occupation)).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = render(<CardStaff {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
