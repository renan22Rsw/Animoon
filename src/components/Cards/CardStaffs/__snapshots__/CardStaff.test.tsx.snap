// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardStaff component should match a snapshot 1`] = `
<div>
  <div
    class=" bg-[#1C1C1C] h-[80px]  w-[350px] m-4 rounded-md lg:w-[381px] 2xl:w-[380px]  2xl:h-[70px]"
  >
    <a
      class="flex items-center text-xs"
      href="/staffs/1"
    >
      <img
        alt="Card-image"
        class="h-20 w-[60px] rounded-sm 2xl:h-[70px] 2xl:w-[55px] "
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="60"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fstaff-image.jpg&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fstaff-image.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fstaff-image.jpg&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
      <span
        class="grid p-2 gap-y-6  grid-rows-2 items-center"
      >
        <h4>
          Masashi Kishimoto
        </h4>
        <p>
          Original Creator
        </p>
      </span>
    </a>
  </div>
</div>
`;
